### ISSUE TITLE
NullReferenceException of .ACL property of a parseObject in Unity

### ISSUE THREAD

**Issue by jeanfabre** [ParsePlatform/Parse-SDK-dotNET#155]
Hi,

 Following @hectorramos (https://twitter.com/hectorramos/status/687344122374389760) suggestion, I am posting an issue here that has been going on for few month already it seems.

 The .ACL property of a parseObject has a bug, it's not accessible. Even the code snippet provided in the guide do not work:

```
var wallPost = new ParseObject("WallPost");
var postACL = new ParseACL();
postACL.SetRoleWriteAccess("Moderators", true);

// the following line will fire and exception:
/*
NullReferenceException: Object reference not set to an instance of an object
Parse.ParseObject.GetFieldForPropertyName (System.String className, System.String propertyName)
Parse.ParseObject.SetProperty[ParseACL] (Parse.ParseACL value, System.String propertyName)
Parse.ParseObject.set_ACL (Parse.ParseACL value)
AclTest.Start () (at Assets/AclTest.cs:23)
*/
wallPost.ACL = postACL;
```
Full working component available below for testing:
https://snipt.net/jeanfabre/parse-bug-repro-test/

There is also a question on StackOveflow: http://stackoverflow.com/questions/33943619/nullreference-when-trying-to-set-acl-property-of-new-parse-object

If it's not a bug: any hints on how to tackle this properly would be great as well as an update of the guide.

Thanks,

 Jean Fabre

**Comment by richardjrossiii** [comment]
This is a duplicate of #90, and was fixed in the 1.7.0 release, which was just accepted into the Unity Asset Store yesterday.

Feel free to re-open this if that doesn't work.

**Comment by jeanfabre** [comment]
Hi,

 I think the problem is actually the package that you submitted that is totally wrong. 

If you import the package currently on the asset store on a fresh project with just my test script, you'll get this:
![screenshot](http://i.imgur.com/G13XWMc.png)



Unless I am totally blind and missing the obvious? :) It seems that the Parse folder is gone, I don't see the 'Parse.Unity.dll', and the actual AssetStore files required for publishing to the asset store. Something is very wrong I think. So 1.7.0 could be featuring the fix indeed ( as mentionned in the log, which is why I rushed out to test it), but I don't think the package contains Parse... :) which could explain the confusion.

One very important thing to do when distributing packages is to always include the version in a text file, currently, there is no way to verify which version is installed in a project, this is prone to misunderstanding and issues like the one we are solving right now.

Regards,

Jean

**Comment by richardjrossiii** [comment]
@jeanfabre It would appear that Unity's asset tools are not properly working at the moment (there are issues with the latest version of Unity not even allowing to compile for asset release, and I wrongfully assumed that rolling back to a known working version of the Unity editor would fix that.

You can still download the latest version from either our [GitHub releases](https://github.com/ParsePlatform/Parse-SDK-dotNET/releases) page, or our [downloads page](http://parse.com/downloads), until we figure out precisely what's wrong with the Unity Asset Store.

**Comment by jeanfabre** [comment]
No worries, thanks for the links on the downloads page, I didn't know Unity support was distributed like that too.

-- It's all good know. 

I'll wait for the Asset store package issue to be resolved before wrapping up an update around 1.7.0 for the PlayMaker Support.

It's great to see that Parse is finally giving proper attention on the support side for Unity, it's reassuring :)

Thanks for your time!

 Jean
